apps/calculation/calculation.cpp

```diff
@@ -52,9 +52,16 @@ void Calculation::reset() {
 void Calculation::setContent(const char * c, Context * context) {
   reset();
   strlcpy(m_inputText, c, sizeof(m_inputText));
-  Evaluation<double> * evaluation = input()->evaluate<double>(*context);
-  evaluation->writeTextInBuffer(m_outputText, sizeof(m_outputText));
-  delete evaluation;
+  if(m_output==nullptr) {
```

Comment:
reset() calls tidy() which ensures that m_output == nullptr
So this patch could be only:
assert(m_output == nullptr);
m_output = input()->evaluate\<double\>(*context);
m_output->writeTextInBuffer(m_outputText, sizeof(m_outputText));